src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,162 @@
+            currentNode.prev = nodeToAdd;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T el : list) {
+            add(el);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> currentNode = getNode(index);
+        Node<T> nodeToSet = new Node<>(currentNode.prev, value, currentNode.next);
+        if (currentNode.next != null) {
+            currentNode.next.prev = nodeToSet;
+        } else {
+            head = nodeToSet;
+        }
+        if (currentNode.prev != null) {
+            currentNode.prev.next = nodeToSet;
+        } else {
+            tail = nodeToSet;
+        }
+        return currentNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> currentNode = getNode(index);
+
+        remove(currentNode.item);
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = getNode(object);
+        if (currentNode == null) {
+            return false;
+        }
+        if (currentNode.next != null) {
+            currentNode.next.prev = currentNode.prev;
+        } else {
+            head = currentNode.prev;
+        }
+        if (currentNode.prev != null) {
+            currentNode.prev.next = currentNode.next;
+        } else {
+            tail = currentNode.next;
+        }
+        size--;
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return tail == null && head == null;
+    }
+
+    private Node<T> getNode(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> foundedNode = tail;
+        for (int i = 0; i < index; i++) {
+            if (foundedNode == null) {
+                throw new IndexOutOfBoundsException();
+            }
+            foundedNode = foundedNode.next;
+        }
+        return foundedNode;
+    }
+
+    private Node<T> getNode(T value) {
+        Node<T> foundedNode = tail;
+        while (!Objects.equals(value, foundedNode.item)) {
+            foundedNode = foundedNode.next;
+            if (foundedNode == null) {
+                return foundedNode;
+            }
+        }
+        return foundedNode;
+    }
+
+    private static class Node<E> {
```

Comment:
Make `Node<T>` an inner class. The purpose of inner classes is to group classes that belong together, which makes your code more readable and maintainable.